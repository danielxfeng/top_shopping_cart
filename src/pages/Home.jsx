import { useEffect, useRef } from "react";
import styles from "../styles/Home.module.css";

// The home page, just a static page with some text.
const Home = () => {
  const ref1 = useRef();
  const ref2 = useRef();

  useEffect(() => {
    const timeout = setTimeout(() => {
      ref1.current.classList.add(styles.ok);
      ref2.current.classList.add(styles.ok);
    }, 500);

    return () => clearTimeout(timeout);
  }, []);

  return (
    <div className={styles.home}>
      <h1 ref={ref1} className={styles.left}>Welcome</h1>
      <section ref={ref2} className={styles.content}>
        <p>
          We are thrilled to welcome you to our store, your one-stop destination
          for all your shopping needs! Whether you're looking for the latest
          electronics, stylish clothing, or unique gifts, we've got you covered
          with a wide range of high-quality products at unbeatable prices.
        </p>
        <p>
          At our store, we strive to provide an exceptional shopping experience
          with an easy-to-navigate store, fast and secure checkout, and a team
          of customer service experts ready to assist you with any questions.
        </p>
        <p>
          Enjoy browsing through our carefully curated collections, and donâ€™t
          forget to check out the exclusive deals available in each category!
        </p>
        <p className={styles.bold}>Happy Shopping!</p>
      </section>
    </div>
  );
};

export default Home;
