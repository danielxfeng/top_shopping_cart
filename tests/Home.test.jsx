import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Home from "../src/pages/Home";

describe("Home page", () => {
  it("renders correct", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it("renders the title", () => {
    render(<Home />);
    expect(screen.getByText("Welcome")).toBeInTheDocument();
  });

  it("renders the content", () => {
    render(<Home />);
    expect(screen.getByText("Happy Shopping!")).toBeInTheDocument();
  });

  it("renders the title", () => {
    render(<Home />);
    expect(screen.getByText(/We are thrilled/i)).toBeInTheDocument();
  });
});
