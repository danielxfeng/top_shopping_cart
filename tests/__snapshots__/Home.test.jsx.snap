// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > renders correct 1`] = `
<div>
  <div
    class="_home_b140de"
  >
    <h1
      class="_left_b140de"
    >
      Welcome
    </h1>
    <section
      class="_content_b140de"
    >
      <p>
        We are thrilled to welcome you to our store, your one-stop destination for all your shopping needs! Whether you're looking for the latest electronics, stylish clothing, or unique gifts, we've got you covered with a wide range of high-quality products at unbeatable prices.
      </p>
      <p>
        At our store, we strive to provide an exceptional shopping experience with an easy-to-navigate store, fast and secure checkout, and a team of customer service experts ready to assist you with any questions.
      </p>
      <p>
        Enjoy browsing through our carefully curated collections, and donâ€™t forget to check out the exclusive deals available in each category!
      </p>
      <p
        class="_bold_b140de"
      >
        Happy Shopping!
      </p>
    </section>
  </div>
</div>
`;
